tasks/filter_team_devices.yml

```diff
@@ -24,12 +24,12 @@
 - debug: var=team_if
 - name: Filter unsupported interface types
   set_fact:
-    otopi_host_net: "{{ host_net | difference(team_if) }}"
+    he_net: "{{ host_net | difference(team_if) }}"
   register: otopi_host_net
```

Comment:
Can't we remove registration and use the fact name directly even in hosted-engine-setup and cockpit-ovirt?

```
- name: Filter unsupported interface types
  set_fact:
    otopi_host_net: "{{ host_net | difference(team_if) }}"
```

And the in https://gerrit.ovirt.org/#/c/110456/2/src/plugins/gr-he-common/network/bridge.py we will use:

```
try:
            validValues = r['otopi_host_net']
        except KeyError:
```

And in https://gerrit.ovirt.org/#/c/110459/2/dashboard/src/helpers/HostedEngineSetup/DefaultValueProvider.js we will use:

```
const interfaces = data.otopi_host_net;
```

Can we use that simpler approach?

Reply:
> Can't we remove registration and use the fact name directly even in hosted-engine-setup and cockpit-ovirt?
> 
> ```
> - name: Filter unsupported interface types
>   set_fact:
>     otopi_host_net: "{{ host_net | difference(team_if) }}"
> ```
> 
> And the in https://gerrit.ovirt.org/#/c/110456/2/src/plugins/gr-he-common/network/bridge.py we will use:
> 
> ```
> try:
>             validValues = r['otopi_host_net']
>         except KeyError:
> ```
> 
> And in https://gerrit.ovirt.org/#/c/110459/2/dashboard/src/helpers/HostedEngineSetup/DefaultValueProvider.js we will use:
> 
> ```
> const interfaces = data.otopi_host_net;
> ```
> 
> Can we use that simpler approach?

I checked that, but it fails in otopi with the following error:

2020-07-24 09:15:17,709+0000 DEBUG otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils._process_output:103 TASK [ovirt.hosted_engine_setup : debug]
2020-07-24 09:15:18,011+0000 DEBUG otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils._process_output:103 otopi_host_net: ['ens2']
2020-07-24 09:15:18,313+0000 INFO otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils._process_output:109 TASK [ovirt.hosted_engine_setup : Failed if only teaming devices are availible]
2020-07-24 09:15:18,515+0000 INFO otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils._process_output:109 skipping: [localhost]
2020-07-24 09:15:18,817+0000 INFO otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils._process_output:109 TASK [ovirt.hosted_engine_setup : Validate selected bridge interface if management bridge does not exist]
2020-07-24 09:15:19,119+0000 INFO otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils._process_output:109 skipping: [localhost]
2020-07-24 09:15:19,421+0000 DEBUG otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils._process_output:103 PLAY RECAP [localhost] : ok: 15 changed: 1 unreachable: 0 skipped: 5 failed: 0
2020-07-24 09:15:19,523+0000 DEBUG otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils.run:215 ansible-playbook rc: 0
2020-07-24 09:15:19,523+0000 DEBUG otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils.run:222 ansible-playbook stdout:
2020-07-24 09:15:19,524+0000 DEBUG otopi.ovirt_hosted_engine_setup.ansible_utils ansible_utils.run:225 ansible-playbook stderr:
2020-07-24 09:15:19,525+0000 DEBUG otopi.plugins.gr_he_common.network.bridge bridge._customization:149 {}
2020-07-24 09:15:19,526+0000 DEBUG otopi.context context._executeMethod:145 method exception
Traceback (most recent call last):
  File "/usr/share/ovirt-hosted-engine-setup/scripts/../plugins/gr-he-common/network/bridge.py", line 151, in _customization
    validValues = r['otopi_host_net']
KeyError: 'otopi_host_net'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/otopi/context.py", line 132, in _executeMethod
    method['method']()
  File "/usr/share/ovirt-hosted-engine-setup/scripts/../plugins/gr-he-common/network/bridge.py", line 154, in _customization
    'No suitable network interfaces were found'
RuntimeError: No suitable network interfaces were found

Reply:
I don't understand enough this otopi code running ansible, but maybe below change could fix it:

```
try:
            validValues = r['ansible_facts']['otopi_host_net']
        except KeyError:
```

It somewhat makes sense, because facts are stored in ansible_facts dictionary and it's only internal ansible shortcut that you can get a value by using key name directly ...

Reply:
Same issue:
2020-07-24 10:30:17,753+0000 DEBUG otopi.plugins.gr_he_common.network.bridge bridge._customization:149 {}
2020-07-24 10:30:17,753+0000 DEBUG otopi.context context._executeMethod:145 method exception
Traceback (most recent call last):
  File "/usr/share/ovirt-hosted-engine-setup/scripts/../plugins/gr-he-common/network/bridge.py", line 151, in _customization
    validValues = r['ansible_facts']['otopi_host_net']
KeyError: 'ansible_facts'

During handling of the above exception, another exception occurred:/var/log/ovirt-hosted-engine-setup/ovirt-hosted-engine-setup-20200724103249-fagxud.log

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/otopi/context.py", line 132, in _executeMethod
    method['method']()
  File "/usr/share/ovirt-hosted-engine-setup/scripts/../plugins/gr-he-common/network/bridge.py", line 154, in _customization
    'No suitable network interfaces were found'
RuntimeError: No suitable network interfaces were found

Reply:
OK, giving up, somebody with deep knowledge of otopi anisble integration will need to investigate, because I think that it's still bad practice to register facts.